include/structopt/parser.hpp

```diff
@@ -57,33 +72,38 @@ struct parser {
     return result;
   }
 
-  bool is_kebab_case(const std::string &next, const std::string &field_name) {
-    bool result = false;
+  bool is_kebab_case(const std::string_view next, const std::string_view field_name) {
     auto maybe_kebab_case = next;
     if (maybe_kebab_case.size() > 1 && maybe_kebab_case[0] == '-') {
       // remove first dash
-      maybe_kebab_case.erase(0, 1);
+      maybe_kebab_case = trim_first_n(maybe_kebab_case, 1);
       if (maybe_kebab_case[0] == '-') {
         // there is a second leading dash
         // remove it
-        maybe_kebab_case.erase(0, 1);
+        maybe_kebab_case = trim_first_n(maybe_kebab_case, 1);
       }
-      std::replace(maybe_kebab_case.begin(), maybe_kebab_case.end(), '-', '_');
-      if (maybe_kebab_case == field_name) {
-        result = true;
+      if (equal_strings_replace_hyphens(maybe_kebab_case, field_name)) {
+        return true;
       }
     }
-    return result;
+    return false;
   }
 
-  bool is_optional_field(const std::string &next, const std::string &field_name) {
-    bool result = false;
-    if (next == "-" + field_name || next == "--" + field_name ||
-        next == "-" + std::string(1, field_name[0]) || is_kebab_case(next, field_name)) {
+  bool is_optional_field(const std::string_view next, const std::string_view field_name) {
+    if (next.size() >= 1 && next[0] == '-' && trim_first_n(next, 1) == field_name) {
+      return true;
+    }
+    if (next.size() >= 2 && next[0] == '-' && next[1] == '-' && trim_first_n(next, 2) == field_name) {
+      return true;
+    }
+    if (next.size() == 2 && next[0] == '-' && next[1] == field_name[0]) {
+      return true;
+    }
```

Comment:
before C++20's `string_view::starts_with` the prefix check was with `rfind` as the following: 
```cpp
if (next.rfind('-', 0) == 0 && next.substr(1) == field_name)
   return true;
if (next.rfind("--", 0) == 0 && next.substr(2) == field_name)
   return true;
if (next == std::data({'-', field_name[0], '\0'})) // #include <initializer_list>
   return true;
   ...
```

Reply:
Done